core/src/main/java/org/hisp/dhis/android/core/arch/api/executors/internal/CoroutineAPICallExecutorImpl.kt

```diff
@@ -127,7 +127,7 @@ internal class CoroutineAPICallExecutorImpl(
     ): P {
         return try {
             databaseAdapter.getCurrentDatabase().useWriterConnection { transactor ->
-                transactor.immediateTransaction {
+                transactor.deferredTransaction {
```

Comment:
Do we need `deferredTransaction` here? I was taking a quick look to `deferredTransaction` and I am not sure... maybe we can have a call for this.